frontend/src/components/dashboard/create/email/EmailTemplate.tsx

```diff
@@ -52,7 +66,15 @@ const EmailTemplate = ({
   const [replyTo, setReplyTo] = useState(
     initialReplyTo === userEmail ? null : initialReplyTo
   )
-  const [from, setFrom] = useState(initialFrom)
+  const {
+    fromName: initialFromName,
+    fromAddress: initialFromAddress,
+  } = initialFrom
+    ? parseFromAddress(initialFrom)
+    : { fromName: '', fromAddress: '' }
```

Comment:
There should never be an empty initialFrom, correct? If that's correct then this should throw an error

Reply:
Turns out that `initialFrom` is undefined when it is a new campaign without a saved template. I've added a comment for this.